File: bank-app/app/src/main/java/com/bankapp/loginScreen/LoginModel.java
package com.bankapp.loginScreen;

import com.bankapp.ErrorResponse;

public class LoginModel {
    public String user;
    public String password;

    public LoginModel(String user, String password) {
        this.user = user;
        this.password = password;
    }
}

class LoginViewModel {
    public UserAccount userAccount;
    public ErrorResponse error;
    public LoginModel loginModel;
    public boolean wrongUser;
    public boolean wrongPassword;
}

class LoginRequest {
    public LoginModel loginModel;
}

File: app/src/main/java/dev/vitorpacheco/solutis/bankapp/loginScreen/LoginModel.kt
package dev.vitorpacheco.solutis.bankapp.loginScreen

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import java.math.BigDecimal

@Parcelize
data class LoginViewModel(
    val user: String? = null,
    val account: UserAccount? = null,
    val errorMessage: String? = null,
    val errorField: UserFormFields? = null
) : Parcelable

data class LoginRequest(var user: String?, var password: String?)

@Parcelize
data class UserAccount(
    val userId: Int,
    val name: String?,
    val bankAccount: String?,
    val agency: String?,
    val balance: BigDecimal?
) : Parcelable

@Parcelize
data class UserError(
    val message: String? = null,
    val field: UserFormFields? = null
) : Parcelable

@Parcelize
data class LoginResponse(
    val userAccount: UserAccount? = null,
    val error: UserError? = null
) : Parcelable

enum class UserFormFields {
    USER, PASSWORD
}